src/app/services/actions/query-action-creators.ts

```diff
@@ -96,6 +96,7 @@ async function createHistory(response: Response, respHeaders: any, query: IQuery
 
 
   const historyItem: IHistoryItem = {
+    index: -1,
```

Comment:
What does this property help with?

Reply:
We are using it for telemetry to check how far a user goes in the history list.